plugins/woocommerce-admin/client/products/fills/product-block-editor-fills.tsx

```diff
@@ -27,16 +20,15 @@ import {
 	AboutTheEditorMenuItem,
 } from '../fills/more-menu-items';
 
-const MoreMenuFill = ( { onClose }: { onClose: () => void } ) => {
-	const [ id ] = useEntityProp( 'postType', 'product', 'id' );
+export type MoreMenuFillProps = { productType?: string; onClose: () => void };
 
-	const { type, status } = useSelect(
-		( select ) => {
-			const { getEntityRecord } = select( 'core' );
-			return getEntityRecord( 'postType', 'product', id ) as Product;
-		},
-		[ id ]
-	);
+export const MoreMenuFill = ( {
+	productType = 'product',
+	onClose,
+}: MoreMenuFillProps ) => {
+	const [ id ] = useEntityProp( 'postType', productType, 'id' );
+	const [ type ] = useEntityProp( 'postType', productType, 'type' );
+	const [ status ] = useEntityProp( 'postType', productType, 'status' );
```

Comment:
Any particular reason you decided to make use of `useEntityProp` here instead of `useSelect` with `getEntityRecord` as before?
It's not a huge deal, but in my mind the `useSelect` may actually be more efficient (but maybe not ).